modules/apps/commerce/commerce-discount-service/src/main/java/com/liferay/commerce/discount/internal/security/permission/resource/CommerceDiscountChannelRelModelResourcePermission.java

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.commerce.discount.internal.security.permission.resource;
+
+import com.liferay.commerce.discount.model.CommerceDiscount;
+import com.liferay.commerce.discount.permission.CommerceDiscountPermission;
+import com.liferay.commerce.discount.service.CommerceDiscountLocalService;
+import com.liferay.commerce.product.model.CommerceChannelRel;
+import com.liferay.commerce.product.service.CommerceChannelRelLocalService;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.security.permission.PermissionChecker;
+import com.liferay.portal.kernel.security.permission.resource.ModelResourcePermission;
+import com.liferay.portal.kernel.security.permission.resource.PortletResourcePermission;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Luca Pellizzon
+ */
+@Component(
+	enabled = false, immediate = true,
+	property = "name=CommerceDiscountChannelRel",
```

Comment:
@marco-leo @dsitu this is my only doubt. Since we didn't want to have a generic model resource permission on the channel rel we decided to add that property name to identify that's going to be used by the discountChannel relation.

Reply:
@riccardo-alberti what's the reason why you guys chose not have go with a generic model? You guys can still do a check specifically on `commerceDiscount`. Just add logic to check the `classNameId` and then based off the `classNameId` you can call the appropriate permission class.

Reply:
@riccardo-alberti I guess what @dsitu says is exactly what we need.
Using the CommerceChannelRel.classNameId to check if we are handling a discount, so that if in the future we will ever need to check anything else we can just add an IF statement.